server/column.go

```diff
@@ -47,9 +55,10 @@ func (column *ColumnInfo) Dump(buffer []byte) []byte {
 	buffer = append(buffer, column.Decimal)
 	buffer = append(buffer, 0, 0)
 
-	if column.DefaultValue != nil {
-		buffer = dumpUint64(buffer, uint64(len(column.DefaultValue)))
-		buffer = append(buffer, column.DefaultValue...)
+	if flags&WithDefaultValue > 0 {
+		// Current we doesn't output defaultValue but reserve defaultValue length bit to make mariadb client happy.
+		// https://dev.mysql.com/doc/internals/en/com-query-response.html#column-definition
+		buffer = dumpUint64(buffer, 0)
```

Comment:
why this change works?

Reply:
because in protocol definition https://dev.mysql.com/doc/internals/en/com-query-response.html#column-definition said.

```
  if command was COM_FIELD_LIST {
lenenc_int     length of default-values
string[$len]   default values
  }
```
So, if it's a response of COM_FIELD_LIST, we should actually add default values part use just like this.

```
+----------------------+-------------------+------------+--------------+--------------+-----------+--------------------+
|                      |                   |            |              |              |           |                    |
|                      |                   |            |              |              |           |                    |
|                      |                   |            |              |              |           |                    |
|                      |                   |            |              |              |           |                    |
|      ....sth we      |  length byte of   |   default  |   default    |   default    |  default  |   ..............   |
|      outputed        |  default values   |   value 1  |   value 2    |   value 3    |  value 4  |                    |
|                      |                   |            |              |              |           |                    |
|                      |                   |            |              |              |           |                    |
|                      |                   |            |              |              |           |                    |
+----------------------+-------------------+------------+--------------+--------------+-----------+--------------------+

```
but In tidb we omit default values(It seems not easy to output that....),  and omit all default values bytes...but mariadb will read `length byte of default values` without condition, after send `COM_FIELD_LIST`. see: https://github.com/MariaDB/server/blob/72b6d01848e56a75349d663bc61bbe71f97a280b/sql-common/client.c#L1519

So at this PR, we don't complete fill all default values but follow the protocol give client an `empty default values`

```
+-------------+--------------+
|             |              |
|             |              |
|             |              |
|  sth we     | length byte  |
|  outputed   | of default   |
|             | values       |
|             |              |
|             |    (0)       |
|             |              |
+-------------+--------------+

```
so mariadb will read length byte without out of index crash.